icu4c/source/i18n/format.cpp

```diff
@@ -70,22 +71,23 @@ FieldPosition::clone() const {
 // default constructor
 
 Format::Format()
-    : UObject()
+    : UObject(), actualLocale(new CharString()), validLocale(new CharString())
 {
```

Comment:
I am troubled by constructors here and elsewhere that don't take a UErrorCode and therefore cannot report failures when the allocations failed (out of memory).

Would it be feasible to track down subclass constructors that invoke the respective super class constructors?

---
Other idea:
- We could try to default-construct these locale string pointers with nullptr
- construct LocaleBased with *references* to the two pointers, so that it can allocate CharString's if needed
- handle nullptr in the getters: return ""
- handle nullptr in the setters: if nullptr then try to allocate

This would also save a little memory when these are not set, or set to the empty string.